Prowjob parser (#2506)
kyma-project/test-infra: development/prowjobparser/README.md

# prowjobparser

## Overview

> Prowjobparser is a helper tool which parse all prowjobs under provided path, match them against provided labels filters and print matching prowjob names to stdout.

Reviewer: ```suggestion
The prowjobparser is a helper tool which parses all ProwJobs under the provided path, matches them against the provided label filters, and prints matching ProwJob names to the standard output.
```

## Usage

### CLI parameters

The prowjobsparser accepts the following command line parameters:

|Parameter | Shorthand | Description |
|-----------|-----------|------------|

> | **configpath** | **-c** | Path to prow config yaml file. |

Reviewer: ```suggestion
| **configpath** | **-c** | Path to the Prow config YAML file. |
```

> | **jobpath** | **-j** | Path to directory containing yaml files with prowjobs. |

Reviewer: ```suggestion
| **jobpath** | **-j** | Path to the directory containing YAML files with ProwJobs. |
```

> | **includepreset** | **-i** | Preset name which must be added to prowjob. Accept multiple param instances. | 

Reviewer: ```suggestion
| **includepreset** | **-i** | Preset name which must be added to ProwJobs. Accepts multiple-parameter instances. | 
```
Please check if the suggestion reflects the original meaning.

> | **excludepreset** | **-e** | Preset name which shouldn't be added to prowjob. Accept multiple param instances. | 

Reviewer: ```suggestion
| **excludepreset** | **-e** | Preset name which must not be added to ProwJobs. Accepts multiple-parameter instances. | 
```

> | **excludepreset** | **-e** | Preset name which must not be added to ProwJobs. Accepts multiple-parameter instances. | 

Reviewer: Shouldn't we also have a section on how to run it?

Author: Added example execution. There is no prerequisites, it's quite simple tool.

# prowjobparser

## Overview

The prowjobparser is a helper tool which parses all ProwJobs under the provided path, matches them against the provided label filters, and prints matching ProwJob names to the standard output.

## Usage

>  `~/go/src/github.com/dekiel/test-infra/development/prowjobparser$ go run main.go -c ../../../../kyma-project/test-infra/prow/config.yaml -j ../../../../kyma-project/test-infra/prow/jobs -i preset-sa-gke-kyma-integration -e preset-sa-kyma-artifacts`

Reviewer: ``````suggestion
To use the prowjobparser, run it with appropriate parameters. See the example:

```bash
go run main.go \
  -c ../../../../kyma-project/test-infra/prow/config.yaml \
  -j ../../../../kyma-project/test-infra/prow/jobs \
  -i preset-sa-gke-kyma-integration \
  -e preset-sa-kyma-artifacts
```
``````
